sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/ImdsRetryStrategy.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation;
+
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.RequestRetryCondition;
+import com.azure.core.http.policy.RetryStrategy;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.function.Predicate;
+import java.util.function.Function;
+
+public class ImdsRetryStrategy implements RetryStrategy {
```

Comment:
I know it's an impl class but I think a javadoc explaining the retry algo would be nice